test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,314 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.path).toEqual(["paths", "/pets", "get", "parameters", "0", "schema", "enum", 1])
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+
+  it("should return an error for a number value inside quotes in a enum number type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "number",
+                in: "query",
+                schema: {
+                  type: "number",
+                  enum: [1, "2", 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.path).toEqual(["paths", "/pets", "get", "parameters", "0", "schema", "enum", 1])
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+  
+  it("should not return an error when all items are number in a enum number type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "number",
+                in: "query",
+                schema: {
+                  type: "number",
+                  enum: [1, 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Array Tests
+
+  it("should return an error for a non array value in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [1, 2, 3]                  
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const secondError = allErrors[1]
+      expect(allErrors.length).toEqual(2)
+      expect(secondError.path).toEqual(["paths", "/pets", "get", "parameters", "0", "schema", "enum", 2])
+      expect(secondError.level).toEqual("warning")
+      expect(secondError.message).toEqual("enum value should conform to its schema's `type`")
```

Comment:
**Change request**: add expectations for each error we're getting back, not just one of them.

(You can also use `expect(...).toInclude` to make the multiple checks look nicer, [here's](https://github.com/swagger-api/swagger-editor/blob/724fac76bf88016bbee397fd8e1a5517f5591afc/test/unit/plugins/validate-semantic/2and3/refs.js#L412-L422) an example. Ping me if you have questions about this pattern.)

Reply:
Include pattern implemented and the other error has been fixed.